cheatsheets/gleam-for-elm-users.md

```diff
@@ -0,0 +1,699 @@
+
+#### Elm
+
+In Elm, type annotations are optionally given on the line above the variable assignment. They can be
+provided in let-blocks but it frequently only provided for top level variables and functions.
+
+```elm
+someList : List Int
+someList = [1, 2, 3]
+```
+
+#### Gleam
+
+In Gleam type annotations can optionally be given when binding variables.
+
+```rust
+let some_list: List(Int) = [1, 2, 3]
+```
+
+### Constants
+
+#### Elm
+
+In Elm, constants can be defined at the top level of the module like any other value and exported if desired and reference from other modules.
+
+```elm
+module Hikers exposing (theAnswer)
+
+theAnswer: Int
+theAnswer =
+    42
+```
+
+#### Gleam
+
+In Gleam constants can be created using the `const` keyword.
+
+```rust
+const the_answer = 42
+
+pub fn main() {
+  the_answer
+}
+```
+
+Gleam constants can be referenced from other modules.
+
+```rust
+// in file other_module.gleam
+pub const the_answer: Int = 42
+```
+
+```rust
+import other_module
+
+fn main() {
+  other_module.the_answer
+}
+```
+
+## Functions
+
+#### Elm
+
+In Elm, functions are defined as declarations that have arguments, or by assigning anonymous functions to variables.
+
+```elm
+sum x y =
+  x + y
+
+mul =
+  \x y -> x * y
+
+mul 3 2
+-- 6
+```
+
+#### Gleam
+
+Gleam's functions are declared using a syntax similar to Rust or JavaScript. Gleam's anonymous functions are declared using the `fn` keyword.
+
+```rust
+pub fn sum(x, y) {
+  x + y
+}
+
+let mul = fn(x, y) { x * y }
+mul(1, 2)
+```
+
+### Function type annotations
+
+#### Elm
+
+In Elm, functions can **optionally** have their argument and return types annotated. These type annotations will always be checked by the compiler and throw a compilation error if not valid. The compiler will still type check your program using type inference if annotations are omitted.
+
+
+```elm
+sum : number -> number -> number
+sum x y = x + y
+
+@spec mul(number, number) :: boolean # no Elixir compile error
+mul : number -> number -> Bool -- Compile error
+mul x y = x * y
+```
+
+#### Gleam
+
+All the same things are true of Gleam though the type annotations go inline in the function
+declaration, rather than above it.
+
+```rust
+pub fn add(x: Int, y: Int) -> Int {
+  x + y
+}
+
+pub fn mul(x: Int, y: Int) -> Bool { // compile error, type mismatch
+  x * y
+}
+```
+
+### Function heads
+
+Neither Elm nor Gleam support multiple function heads like Erlang, Elixir or Haskell. Both languages
+would expect a case-statement to be used to handle variations in data being supplied to the
+function.
+
+### Function overloading
+
+Neither Elm nor Gleam support function overloading.
+
+### Referencing functions
+
+Both Elm and Gleam have a single namespace for value and functions within a module, so there
+is no need for a special syntax to assign a module function to a variable.
+
+### Calling anonymous functions
+
+Neither Elm nor Gleam require special syntax for calling anonymous functions.
+
```

Comment:
Maybe we can remove this bit? Not sure Elm programmers would be expecting these features. What do you think?

```suggestion
```

Reply:
Yeah, makes total sense. This a hang over from the original Elixir layout and it seemed odd to delete valid information but I can see it is potentially confusing if someone knows little about these things.